File: packages/server-go/internal/models/workflow.go
package models

import "time"

// Workflow represents a BPMN workflow
type Workflow struct {
	ID          string    `json:"id" db:"id"`
	Name        string    `json:"name" db:"name"`
	Description string    `json:"description,omitempty" db:"description"`
	BpmnXML     string    `json:"bpmnXml" db:"bpmn_xml"`
	Version     string    `json:"version" db:"version"`
	Status      string    `json:"status" db:"status"`
	CreatedBy   string    `json:"createdBy,omitempty" db:"created_by"`
	CreatedAt   time.Time `json:"createdAt" db:"created_at"`
	UpdatedAt   time.Time `json:"updatedAt" db:"updated_at"`
}

// WorkflowStatus constants
const (
	StatusDraft    = "draft"
	StatusActive   = "active"
	StatusInactive = "inactive"
	StatusArchived = "archived"
)

File: server-nodejs/src/models/workflow.ts
export interface Workflow {
  id: string;
  name: string;
  description?: string;
  bpmnXml: string;
  version: string;
  status: WorkflowStatus;
  createdBy?: string;
  createdAt: Date;
  updatedAt: Date;
}

export enum WorkflowStatus {
  DRAFT = 'draft',
  ACTIVE = 'active',
  INACTIVE = 'inactive',
  ARCHIVED = 'archived',
}

export interface Node {
  id: string;
  parentId?: string;
  name: string;
  type: number;
  incomingSequenceFlowIds: string[];
  outgoingSequenceFlowIds: string[];
  businessApiUrl?: string;
  attachedNodeId?: string;
  canFallback: boolean;
}

export interface SequenceFlow {
  id: string;
  name: string;
  sourceNodeId: string;
  targetNodeId: string;
  conditionExpression?: string;
  priority: number;
}

export interface Message {
  id: string;
  name: string;
}

export interface VariableDeclaration {
  name: string;
  type: number;
}

export interface WorkflowDefinition {
  nodes: Record<string, Node>;
  startEvents: string[];
  endEvents: string[];
  sequenceFlows: Record<string, SequenceFlow>;
  messages: Record<string, Message>;
  variableDeclarations: VariableDeclaration[];
  adjacencyList: Record<string, string[]>;
  reverseAdjacencyList: Record<string, string[]>;
}
